addons/stock/models/stock_move.py

```diff
@@ -564,6 +564,8 @@ def write(self, vals):
         # messages according to the state of the stock.move records.
         receipt_moves_to_reassign = self.env['stock.move']
         move_to_recompute_state = self.env['stock.move']
+        if 'product_uom' in vals and any(move.state == 'done' for move in self):
```

Comment:
Can we try to remove it for 15.0?

Reply:
So.. r+ as is? (discussed offline that this isn't really doable)